lib/index.js

```diff
@@ -840,8 +840,12 @@ Stream.prototype.destroy = function () {
     _(this._consumers).each(function (consumer) {
         self._removeConsumer(consumer);
     });
+    this._observers.length = 0;
```

Comment:
Should this also loop through each observer and do a `_removeObserver`? Otherwise, downstream observers will forever hold a reference to this stream.